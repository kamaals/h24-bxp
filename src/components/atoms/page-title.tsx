import React from "react";

function PageTitle({ title }: { title: string }) {
  return (
    <h1 className={"text-2xl md:text-3xl lg:text-4xl font-bold font-sans"}>
      {title}
    </h1>
  );
}

export default PageTitle;
