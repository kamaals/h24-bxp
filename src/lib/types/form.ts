import { useCreateCategoryMutation } from "@/lib/store/api/categoryServices";

export type GenericMutationType = typeof useCreateCategoryMutation;
