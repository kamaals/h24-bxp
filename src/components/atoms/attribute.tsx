import React from "react";
import { AttributeType, AttributeTypes } from "@/lib/types/product";
import { cn } from "@/lib/utils";

function Attribute({ attribute }: { attribute: AttributeType }) {
  const type = attribute.type as AttributeTypes;
  const bgClassName = () => {
    switch (type) {
      case "boolean":
        return "bg-rose-100";
      case "number":
        return "bg-violet-100";
      case "tag":
        return "bg-lime-100";
      case "text":
        return "bg-emerald-100";
      case "url":
        return "bg-cyan-100";
      default:
        return "bg-white";
    }
  };

  return (
    <span className={cn("rounded-lg text-xs p-2 inline-block", bgClassName())}>
      <span className={"flex gap-1"}>
        <span className="font-bold">{attribute.code}: </span>
        <span>{attribute.name}</span>
      </span>
      <span>{attribute.type}</span>
    </span>
  );
}

export default Attribute;
