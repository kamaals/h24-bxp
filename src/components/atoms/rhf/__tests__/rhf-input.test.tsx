import { describe, expect, it, beforeAll } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import RHFInput from "../rhf-input";
import { useForm } from "react-hook-form";
import { Form } from "@/components/atoms/form";
import React from "react";

const FormProvider = ({
  children,
  value,
  error,
}: {
  children: React.ReactNode;
  value?: string;
  error?: string;
}) => {
  const form = useForm<{ name: string }>({
    defaultValues: { name: value },
  });

  React.useEffect(() => {
    if (error) {
      form.setError("name", { message: error });
    }
  }, [error, form]);

  return <Form {...form}>{children}</Form>;
};

describe("RHFInput", () => {
  beforeAll(() => {
    cleanup();
  });

  it("should match the snapshot", () => {
    const result = render(
      <FormProvider value={"Alpha"}>
        <RHFInput name={"name"} label={"Name"} />
      </FormProvider>,
    );
    expect(result).toMatchSnapshot();
  });

  it("should render", () => {
    const { container } = render(
      <FormProvider>
        <RHFInput name={"name"} label={"Name"} />
      </FormProvider>,
    );
    const nameInput = container.querySelector(
      'input[name="name"]',
    ) as HTMLInputElement;
    expect(nameInput).toBeInTheDocument();
  });

  it("should trigger error", () => {
    render(
      <FormProvider error={"Not valid"}>
        <RHFInput name={"name"} label={"Name"} />
      </FormProvider>,
    );
    const errorText = screen.getByText("Not valid");
    expect(errorText).toBeInTheDocument();
  });
});
