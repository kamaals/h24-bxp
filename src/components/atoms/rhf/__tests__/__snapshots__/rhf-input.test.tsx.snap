// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RHFInput > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <div>
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r0»-form-item"
          >
            Name
          </label>
          <p
            class="text-muted-foreground text-xs"
            data-slot="form-description"
            id="«r0»-form-item-description"
            style="margin-top: -.125rem;"
          />
        </div>
        <div
          aria-describedby="«r0»-form-item-description"
          aria-invalid="false"
          class="space-y-1"
          data-slot="form-control"
          id="«r0»-form-item"
        >
          <input
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="input"
            data-testid="name"
            name="name"
            value="Alpha"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid gap-2"
      data-slot="form-item"
    >
      <div>
        <label
          class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
          data-error="false"
          data-slot="form-label"
          for="«r0»-form-item"
        >
          Name
        </label>
        <p
          class="text-muted-foreground text-xs"
          data-slot="form-description"
          id="«r0»-form-item-description"
          style="margin-top: -.125rem;"
        />
      </div>
      <div
        aria-describedby="«r0»-form-item-description"
        aria-invalid="false"
        class="space-y-1"
        data-slot="form-control"
        id="«r0»-form-item"
      >
        <input
          class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="input"
          data-testid="name"
          name="name"
          value="Alpha"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
